src/libs/tools/tests/testtool_backendbuilder.cpp

```diff
@@ -546,3 +546,101 @@ TEST (BackendBuilder, resolveDoubleRecommends)
 	EXPECT_EQ (bb.cbegin ()[1], PluginSpec ("a"));
 	EXPECT_EQ (bb.cbegin ()[2], PluginSpec ("c"));
 }
+
+static int checkconfLookup (ckdb::Key * errorKey, ckdb::KeySet * config)
+{
+	ckdb::Key * k = ckdb::ksLookupByName (config, "/a", 0);
+	if (k)
+	{
+		return 0;
+	}
+	return -1;
+}
+
+TEST (BackendBuilder, checkconfOkNoChange)
+{
+	using namespace kdb;
+	using namespace kdb::tools;
+	std::shared_ptr<MockPluginDatabase> mpd = std::make_shared<MockPluginDatabase> ();
+	mpd->data[PluginSpec ("checkconf1")]["provides"] = "test123";
+	mpd->setCheckconfFunction (checkconfLookup);
+	BackendBuilderInit bbi (mpd);
+	BackendBuilder bb (bbi);
+	PluginSpec spec ("checkconf1");
+	KeySet pluginConfig;
+	Key a;
+	a.setName ("user/a");
+	a.setString ("abc");
+	pluginConfig.append (a);
+	spec.appendConfig (pluginConfig);
+	bb.addPlugin (spec);
+}
+
+TEST (BackendBuilder, checkconfNotOKmissing)
+{
+	using namespace kdb;
+	using namespace kdb::tools;
+	std::shared_ptr<MockPluginDatabase> mpd = std::make_shared<MockPluginDatabase> ();
+	mpd->data[PluginSpec ("checkconf3")]["c"] = "something";
+	mpd->setCheckconfFunction (checkconfLookup);
+	BackendBuilderInit bbi (mpd);
+	BackendBuilder bb (bbi);
+	EXPECT_THROW (bb.addPlugin (PluginSpec ("checkconf3")), PluginConfigInvalid);
+}
+
+static int checkconfAppend (ckdb::Key * errorKey, ckdb::KeySet * config)
+{
+	ckdb::Key * k = ckdb::ksLookupByName (config, "user/b", 0);
+	if (!k)
+	{
+		ckdb::ksAppendKey (config, ckdb::keyNew ("user/b", KEY_VALUE, "test", KEY_END));
```

Comment:
AppendKey will overwrite the key if it exists. The name of the function suggests that it will always append.